### PR TITLE
[node] swap `undici.fetch` for `undici.request` in `serverless-handler.mts` 

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
In a recent undici update, setting the `host` header for fetch requests became invalid (https://github.com/nodejs/undici/pull/2322). 

We relied on this in order to proxy serverless dev server requests via `@vercel/node`. 

This PR replaces the usage of `undici.fetch` with `undici.request`. 

It is blocked by an `undici` type change: https://github.com/nodejs/undici/pull/2373

### CODE REVIEW

**Comment by martinez-hugo** [comment]
@Ethan-Arrowood I think that error in `Lint / Lint (pull_request)` workflow come from type definitions of argument in `@edge-runtime/node-utils`.

In fact, the `toToReadable` ask a first argument `webStream` which require a `ReadableStream` type argument and not a `Readable` type arg.

Si I've made this PR : https://github.com/vercel/edge-runtime/pull/672

File: packages/node/src/edge-functions/edge-handler.mts [hunk 1]
```diff
@@ -23,6 +23,9 @@ if (!NODE_VERSION_MAJOR) {
   );
 }
 
+// @ts-expect-error
```
Comment (EndangeredMassa) [inline packages/node/src/edge-functions/edge-handler.mts]: **suggestion (non-blocking):** add note about why this error is ignored‏


```suggestion
// @ts-expect-error - depends on https://github.com/nodejs/undici/pull/2373
```

**Comment by Ethan-Arrowood** [comment]
Some tests were failing to do log incident yesterday. re-running now

File: packages/node/src/serverless-functions/serverless-handler.mts [hunk 2]
```diff
@@ -103,43 +86,40 @@ export async function createServerlessEventHandler(
...
-          if (key !== 'transfer-encoding') {
-            // transfer-encoding is only for streaming response
-            clonedHeaders.push([key, value]);
-          }
-        }
-        clonedHeaders.push(['content-length', String(Buffer.byteLength(body))]);
-        headers = new Headers(clonedHeaders);
-      }
+      body = Buffer.from(await response.body.arrayBuffer());
+      // console.log('body', body.toString('utf-8'));
```
Comment (cb1kenobi) [inline packages/node/src/serverless-functions/serverless-handler.mts]: **question (non-blocking):** ‏Is there any reason to keep this commented out block?

Comment (Ethan-Arrowood) [reply packages/node/src/serverless-functions/serverless-handler.mts]: just removed :)

**Review by EndangeredMassa — DISMISSED** [review]

**Review by trek — DISMISSED** [review]

**Review by EndangeredMassa — APPROVED** [review]

**Pull request merged**